### ISSUE TITLE
Macro State is not preserved between evaluations

### ISSUE THREAD

**Issue by kunstmusik** [csound/csound#896]
When live coding, any macros that are defined during a compilation pass are lost and not available in future compilations.  For example, if the following is defined in a file called myfile.udo:

```csound
#ifndef MYFILE_UDO
#define MYFILE_UDO #1#
;; one-time code
#endif
```

and if you #include it multiple times in a CSD and start csound with that CSD, the one-time code will truly only get run once.  However, if you are live coding, and if you evaluate the file by #including the file or evaluating the contents, it will run every time as the definition of MYFILE_UDO lost after each evaluation. 

What I expected was that the state of macros be maintained between code evaluations and that the above code run only once, whether it is in the context of single-pass evaluation and render or live coding with multiple evaluations.

**Comment by vlazzarini** [comment]
In my opinion, this is not a bug. Macros are not part of the language, they are preprocessor definitions that only apply to a given compilation.
By changing this, we will be introducing them as names/constants that will have to be kept within the engine and I do not think this is the right semantics.

**Comment by jpffitch** [comment]
Current version in github has persistent macros for orc sections.  Not done scores as they are once only at present.

Try it and see if it is what you want.

**Comment by jpffitch** [comment]
Anyone tried it yet?
